codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -906,6 +916,21 @@ public DefaultStream createStream(int streamId) throws Http2Exception {
         }
 
         @Override
+        public DefaultStream createIdleStream(int streamId) throws Http2Exception {
+            return createStream(streamId, IDLE);
+        }
+
+        @Override
+        public DefaultStream createStream(int streamId, boolean halfClosed) throws Http2Exception {
```

Comment:
The majority of the time we are using this method...is it only in rare cases (PRIORITY frames) does it make sense to not open the stream and in these cases `createIdleStream` should be used.   This allows the stream object's state to be consistent when the listener methods are invoked.